kotlin/reflect/src/main/java/com/squareup/moshi/kotlin/reflect/KotlinJsonAdapter.kt

```diff
@@ -133,46 +146,59 @@ internal class KotlinJsonAdapter<T>(
     writer.endObject()
   }
 
-  override fun toString() = "KotlinJsonAdapter(${constructor.returnType})"
+  override fun toString() = "KotlinJsonAdapter(${constructor.type.canonicalName})"
 
   data class Binding<K, P>(
-    val name: String,
-    val jsonName: String?,
-    val adapter: JsonAdapter<P>,
-    val property: KProperty1<K, P>,
-    val parameter: KParameter?,
-    val propertyIndex: Int
+      val name: String,
+      val jsonName: String?,
+      val adapter: JsonAdapter<P>,
+      val property: PropertyData,
```

Comment:
(I wonder if there’s a more descriptive suffix than Data...)

Reply:
this is what we called this in KotlinPoet-metadata, which I mentally modeled some of this off of. Open to suggestions!

Reply:
f3e3003